tests/drift/test_drift.py

```diff
@@ -584,3 +596,81 @@ def test_result_comparison_to_cbpe_plots_raise_no_exceptions(sample_drift_data):
         _ = result.compare(result2).plot()
     except Exception as exc:
         pytest.fail(f"an unexpected exception occurred: {exc}")
+
+
+def test_univariate_drift_result_extension_attributes_should_be_in_place(sample_drift_data):
+    ref_data = sample_drift_data.loc[sample_drift_data['period'] == 'reference']
+    calc = UnivariateDriftCalculator(
+        column_names=['f1', 'f2', 'f3', 'f4'],
+        timestamp_column_name='timestamp',
+        continuous_methods=['kolmogorov_smirnov'],
+        categorical_methods=['chi2'],
+    ).fit(ref_data)
+    result = calc.calculate(data=sample_drift_data)
+
+    # Data model is set up so f1, f2 are continuous and f3, f4 are categorical
+    assert result.continuous_column_names == ['f1', 'f2']
+    assert result.categorical_column_names == ['f3', 'f4']
+    assert result.timestamp_column_name == 'timestamp'
+    assert result.continuous_method_names == ['kolmogorov_smirnov']
+    assert result.categorical_method_names == ['chi2']
+    assert [m.column_name for m in result.continuous_methods] == ['kolmogorov_smirnov']
+    assert [m.column_name for m in result.categorical_methods] == ['chi2']
+    assert sorted(m.column_name for m in result.methods) == sorted(('chi2', 'kolmogorov_smirnov'))
+
+
+def test_univariate_drift_result_filter_should_preserve_data_with_default_args(univariate_drift_result):
+    filtered_result = univariate_drift_result.filter()
+    assert filtered_result.data.equals(univariate_drift_result.data)
+    assert filtered_result.continuous_column_names == univariate_drift_result.continuous_column_names
+    assert filtered_result.categorical_column_names == univariate_drift_result.categorical_column_names
+    assert filtered_result.timestamp_column_name == univariate_drift_result.timestamp_column_name
+    assert filtered_result.continuous_method_names == univariate_drift_result.continuous_method_names
+    assert filtered_result.categorical_method_names == univariate_drift_result.categorical_method_names
+    assert filtered_result.continuous_methods == univariate_drift_result.continuous_methods
+    assert filtered_result.categorical_methods == univariate_drift_result.categorical_methods
+
+
+def test_unvariate_drift_result_filter_metrics(univariate_drift_result):
+    filtered_result = univariate_drift_result.filter(methods=['chi2'])
+    metrics = tuple(set(metric for (_, metric, _) in filtered_result.data.columns if metric != 'chunk'))
+    assert metrics == ('chi2',)
+    assert filtered_result.data.shape[0] == univariate_drift_result.data.shape[0]
+
+    assert filtered_result.continuous_column_names == ['f1', 'f2']
```

Comment:
I've set up the test case to match the current filtering behaviour, but I think this may be a bug. The applied filter removes all continuous methods, leaving only categorical methods. The filtered `Result` correctly reports there are no continuous methods, but it retains continuous column names from the original `Result` object - even when those are not available.

Reply:
Confirmed as a bug during engineering review. The `Result.filter` method and test cases have been adjusted to resolve this.